jcl/src/java.base/share/classes/jdk/crac/Core.java

```diff
@@ -41,7 +41,12 @@ public class Core {
 	 */
 	public static Context<Resource> getGlobalContext() {
 		if ((globalContext == null) && InternalCRIUSupport.isCRaCSupportEnabled()) {
-			globalContext = new CRIUSupportContext<>();
+			try {
+				globalContext = new CRIUSupportContext<>();
+			} catch (IllegalArgumentException e) {
```

Comment:
Is there a sample stack trace showing the origin of this `IllegalArgumentException`?

Reply:
```
	at openj9/internal/criu/InternalCRIUSupport.setImageDir(Ljava/nio/file/Path;)Lopenj9/internal/criu/InternalCRIUSupport; (InternalCRIUSupport.java:420)
	at openj9/internal/criu/InternalCRIUSupport.<init>(Ljava/nio/file/Path;)V (InternalCRIUSupport.java:298)
	at jdk/crac/CRIUSupportContext.<init>()V (Core.java:76)
	at jdk/crac/Core.getGlobalContext()Ljdk/crac/Context; (Core.java:44)
	at jdk/internal/reflect/NativeMethodAccessorImpl.invoke0(Ljava/lang/reflect/Method;Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (Native Method)
	at jdk/internal/reflect/NativeMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (NativeMethodAccessorImpl.java:77)
	at jdk/internal/reflect/DelegatingMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (DelegatingMethodAccessorImpl.java:43)
	at java/lang/reflect/Method.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (Method.java:574)
	at org/crac/Core$Compat.<init>(Ljava/lang/String;)V (Core.java:132)
	at org/crac/Core.loadCompat(Ljava/lang/String;)Lorg/crac/Core$Compat; (Core.java:187)
	at org/crac/Core.<clinit>()V (Core.java:205)
	at org/springframework/context/support/DefaultLifecycleProcessor$CracDelegate.registerResource()Ljava/lang/Object; (DefaultLifecycleProcessor.java:521)
	at org/springframework/context/support/DefaultLifecycleProcessor.<init>()V (DefaultLifecycleProcessor.java:122)
	at org/springframework/boot/autoconfigure/context/LifecycleAutoConfiguration.defaultLifecycleProcessor(Lorg/springframework/boot/autoconfigure/context/LifecycleProperties;)Lorg/springframework/context/support/DefaultLifecycleProcessor; (LifecycleAutoConfiguration.java:43)
	at jdk/internal/reflect/NativeMethodAccessorImpl.invoke0(Ljava/lang/reflect/Method;Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (Native Method)
	at jdk/internal/reflect/NativeMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (NativeMethodAccessorImpl.java:77)
	at jdk/internal/reflect/DelegatingMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (DelegatingMethodAccessorImpl.java:43)
	at java/lang/reflect/Method.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (Method.java:574)
	at org/springframework/beans/factory/support/SimpleInstantiationStrategy.instantiate(Lorg/springframework/beans/factory/support/RootBeanDefinition;Ljava/lang/String;Lorg/springframework/beans/factory/BeanFactory;Ljava/lang/Object;Ljava/lang/reflect/Method;[Ljava/lang/Object;)Ljava/lang/Object; (SimpleInstantiationStrategy.java:140)
	at org/springframework/beans/factory/support/ConstructorResolver.instantiate(Ljava/lang/String;Lorg/springframework/beans/factory/support/RootBeanDefinition;Ljava/lang/Object;Ljava/lang/reflect/Method;[Ljava/lang/Object;)Ljava/lang/Object; (ConstructorResolver.java:647)
	at org/springframework/beans/factory/support/ConstructorResolver.instantiateUsingFactoryMethod(Ljava/lang/String;Lorg/springframework/beans/factory/support/RootBeanDefinition;[Ljava/lang/Object;)Lorg/springframework/beans/BeanWrapper; (ConstructorResolver.java:639)
	at org/springframework/beans/factory/support/AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(Ljava/lang/String;Lorg/springframework/beans/factory/support/RootBeanDefinition;[Ljava/lang/Object;)Lorg/springframework/beans/BeanWrapper; (AbstractAutowireCapableBeanFactory.java:1334)
	at org/springframework/beans/factory/support/AbstractAutowireCapableBeanFactory.createBeanInstance(Ljava/lang/String;Lorg/springframework/beans/factory/support/RootBeanDefinition;[Ljava/lang/Object;)Lorg/springframework/beans/BeanWrapper; (AbstractAutowireCapableBeanFactory.java:1164)
	at org/springframework/beans/factory/support/AbstractAutowireCapableBeanFactory.doCreateBean(Ljava/lang/String;Lorg/springframework/beans/factory/support/RootBeanDefinition;[Ljava/lang/Object;)Ljava/lang/Object; (AbstractAutowireCapableBeanFactory.java:561)
	at org/springframework/beans/factory/support/AbstractAutowireCapableBeanFactory.createBean(Ljava/lang/String;Lorg/springframework/beans/factory/support/RootBeanDefinition;[Ljava/lang/Object;)Ljava/lang/Object; (AbstractAutowireCapableBeanFactory.java:521)
	at org/springframework/beans/factory/support/AbstractBeanFactory.lambda$doGetBean$0(Ljava/lang/String;Lorg/springframework/beans/factory/support/RootBeanDefinition;[Ljava/lang/Object;)Ljava/lang/Object; (AbstractBeanFactory.java:325)
	at org/springframework/beans/factory/support/AbstractBeanFactory$$Lambda$330/0x000000009d330428.getObject()Ljava/lang/Object; (Bytecode PC: 16)
	at org/springframework/beans/factory/support/DefaultSingletonBeanRegistry.getSingleton(Ljava/lang/String;Lorg/springframework/beans/factory/ObjectFactory;)Ljava/lang/Object; (DefaultSingletonBeanRegistry.java:234)
	at org/springframework/beans/factory/support/AbstractBeanFactory.doGetBean(Ljava/lang/String;Ljava/lang/Class;[Ljava/lang/Object;Z)Ljava/lang/Object; (AbstractBeanFactory.java:323) (Compiled Code)
	at org/springframework/beans/factory/support/AbstractBeanFactory.getBean(Ljava/lang/String;)Ljava/lang/Object; (AbstractBeanFactory.java:199)
	at org/springframework/beans/factory/support/DefaultListableBeanFactory.preInstantiateSingletons()V (DefaultListableBeanFactory.java:975)
	at org/springframework/context/support/AbstractApplicationContext.finishBeanFactoryInitialization(Lorg/springframework/beans/factory/config/ConfigurableListableBeanFactory;)V (AbstractApplicationContext.java:959)
	at org/springframework/context/support/AbstractApplicationContext.refresh()V (AbstractApplicationContext.java:624)
	at org/springframework/boot/web/servlet/context/ServletWebServerApplicationContext.refresh()V (ServletWebServerApplicationContext.java:146)
	at org/springframework/boot/SpringApplication.refresh(Lorg/springframework/context/ConfigurableApplicationContext;)V (SpringApplication.java:754)
	at org/springframework/boot/SpringApplication.refreshContext(Lorg/springframework/context/ConfigurableApplicationContext;)V (SpringApplication.java:456)
	at org/springframework/boot/SpringApplication.run([Ljava/lang/String;)Lorg/springframework/context/ConfigurableApplicationContext; (SpringApplication.java:334)
	at org/springframework/boot/SpringApplication.run([Ljava/lang/Class;[Ljava/lang/String;)Lorg/springframework/context/ConfigurableApplicationContext; (SpringApplication.java:1354)
	at org/springframework/boot/SpringApplication.run(Ljava/lang/Class;[Ljava/lang/String;)Lorg/springframework/context/ConfigurableApplicationContext; (SpringApplication.java:1343)
	at com/example/Application.main([Ljava/lang/String;)V (Application.java:10)
	at jdk/internal/reflect/NativeMethodAccessorImpl.invoke0(Ljava/lang/reflect/Method;Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (Native Method)
	at jdk/internal/reflect/NativeMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (NativeMethodAccessorImpl.java:77)
	at jdk/internal/reflect/DelegatingMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (DelegatingMethodAccessorImpl.java:43)
	at java/lang/reflect/Method.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; (Method.java:574)
	at org/springframework/boot/loader/launch/Launcher.launch(Ljava/lang/ClassLoader;Ljava/lang/String;[Ljava/lang/String;)V (Launcher.java:91)
	at org/springframework/boot/loader/launch/Launcher.launch([Ljava/lang/String;)V (Launcher.java:53)
	at org/springframework/boot/loader/launch/JarLauncher.main([Ljava/lang/String;)V (JarLauncher.java:58)
```

Reply:
But you wont see it on a normal run because Spring/Crac will eat the exception

Reply:
The default response when there is a failure is to start in standard mode, so it ends up not working and there is no info indicating why.